### PR TITLE
Adapting to `ethersjs`

### LINKED ISSUE #57: The `cli` should be able to govern the nevermined config contract

### PR DESCRIPTION
**Pull request by aaitor**
- adding command
- Adapting to `ethersjs`

## Description

Add a description of your changes here.

## Is this PR related with an open issue?

Related to Issue #

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes
- [ ] Documentation

### CODE REVIEW

File: src/commands/nfts/deployNft.ts [hunk 1]
```diff
@@ -39,37 +53,45 @@ export const deployNft = async (
...
+    const methodSignature = getSignatureOfMethod(
+      contractInstance,
+      'initialize',
+      args
+    )
+    const contract = contractInstance.connect(signer)
+    const transactionResponse: TransactionResponse = await contract[
+      methodSignature
+    ](...args)
+    // ](...args, sendOptions)
```
Comment (mrsmkl) [inline src/commands/nfts/deployNft.ts]: clean up?

File: src/utils/utils.ts [hunk 2]
```diff
@@ -55,14 +55,14 @@ export const loginMarketplaceApi = async (
 
 export const loadContract = (
   config: Config,
-  abi: AbiItem[] | AbiItem,
+  abi: string,
   address: string
 ): Contract => {
   const web3 = Web3Provider.getWeb3(config)
-  web3.setProvider(config.web3Provider)
+  // web3.setProvider(config.web3Provider)
```
Comment (mrsmkl) [inline src/utils/utils.ts]: cleanup?

File: src/utils/utils.ts [hunk 3]
```diff
@@ -55,14 +55,14 @@ export const loginMarketplaceApi = async (
...
 
-  // @ts-ignore
-  return new web3.eth.Contract(abi, address)
+  return new ethers.Contract(address, abi, web3)
+  //return new web3.eth.Contract(abi, address)
```
Comment (mrsmkl) [inline src/utils/utils.ts]: cleanup?

**Review by mrsmkl — APPROVED** [review]
Looking good!

File: README.md [hunk 4]
```diff
@@ -114,34 +114,51 @@ The CLI has pre-configured some Nevermined environments. You can check what is p
...
+        Gateway: http://localhost:8030
+        Marketplace API: http://172.17.0.1:3100
+        Faucet: http://localhost:3001
 
- spree:
-   Gateway: http://localhost:8030
-   Metadata API: http://nevermined-metadata:5000
-   Faucet: http://localhost:3001
+
+ testnet:
```
Comment (r-marques) [inline README.md:127]: I would probably name this `mumbai`

File: README.md [hunk 5]
```diff
@@ -114,34 +114,51 @@ The CLI has pre-configured some Nevermined environments. You can check what is p
...
+        Marketplace API: https://marketplace-api.autonomies.test.nevermined.rocks
+        Faucet: https://faucet.mumbai.public.nevermined.rocks
+
 
 ```
 
 A part of these networks you can connect to any other network using the environment variables described above.
 
-When you want to connect and interact with a different network of the default (`spree`), make sure you have exported the `NODE_URL` environment variable connected to the right environment, and pass the `-n NETWORK_NAME` parameter to your commands.
+When you want to connect and interact with a different network of the default (`testnet`), make sure you have exported the `NODE_URL` environment variable connected to the right environment, and pass the `-n NETWORK_NAME` parameter to your commands.
```
Comment (r-marques) [inline README.md:161]: here too

File: resources/commands.json [hunk 6]
```diff
@@ -9,7 +9,7 @@
 		{
 			"name": "network",
 			"alias": "n",
-			"default": "spree",
+			"default": "testnet",
```
Comment (r-marques) [inline resources/commands.json:12]: here too

File: resources/networks.json [hunk 7]
```diff
@@ -35,7 +35,7 @@
 		"nativeToken": "MATIC",
 		"networkId": "80001",
 		"networkName": "mumbai",
-		"contractsVersion": "2.0.0",
+		"contractsVersion": "2.0.3",
```
Comment (r-marques) [inline resources/networks.json:58]: should this now be 2.0.4?

File: src/cli.ts [hunk 8]
```diff
@@ -91,9 +91,9 @@ const cmdHandler = async (
     )
   }
 
-  const accounts = await nvm.accounts.list()
-  const userAccount = findAccountOrFirst(accounts, argv.account)
-
+  // const accounts = await nvm.accounts.list()
+  // const userAccount = findAccountOrFirst(accounts, argv.account)
```
Comment (r-marques) [inline src/cli.ts]: cleanup

File: src/commands/accounts/list.ts [hunk 9]
```diff
@@ -50,16 +48,15 @@ export const accountsList = async (
 
   const loadedAccounts = await Promise.all(
     accounts.map(async (a, index) => {
-      const ethBalance = utils.fromWei(
-        (await a.getEtherBalance()).toString(),
-        'ether'
+      const ethBalance = ethers.utils.parseEther(
```
Comment (r-marques) [inline src/commands/accounts/list.ts]: you can just call `BigNumber.parseEther`. trying to remove the need to call ethers directly when working with the sdk

File: src/commands/assets/getAsset.ts [hunk 10]
```diff
@@ -3,7 +3,7 @@ import { StatusCodes } from '../../utils'
 import chalk from 'chalk'
 import readline from 'readline'
 import { Logger } from 'log4js'
-import { makeKeyTransfer } from '@nevermined-io/nevermined-sdk-js/dist/node/utils'
+// import { makeKeyTransfer } from '@nevermined-io/nevermined-sdk-js/dist/node/utils'
```
Comment (r-marques) [inline src/commands/assets/getAsset.ts:6]: cleanup

File: src/commands/assets/getAsset.ts [hunk 11]
```diff
@@ -21,16 +21,17 @@ export const getAsset = async (
...
-    account.babyX = key.x
-    account.babyY = key.y
-    account.babySecret = password
-  }
+  // if (password) {
+  //   const key = await keyTransfer.secretToPublic(keyTransfer.makeKey(password))
+  //   account.babyX = key.x
+  //   account.babyY = key.y
+  //   account.babySecret = password
+  // }
```
Comment (r-marques) [inline src/commands/assets/getAsset.ts:34]: maybe create a new issue to address this?

File: src/commands/network/networkGetConfig.ts [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+} from '../../utils'
+import { Logger } from 'log4js'
+import { Account, Nevermined } from '@nevermined-io/nevermined-sdk-js'
+import { ExecutionOutput } from '../../models/ExecutionOutput'
+import * as fs from 'fs'
+import chalk from 'chalk'
+import { ConfigEntry } from '../../models/ConfigDefinition'
+import { ethers } from 'ethers'
+
+export const networkGetConfig = async (
```
Comment (r-marques) [inline src/commands/network/networkGetConfig.ts:14]: maybe rename this to `neverminedContractConfig` or `governanceConfig`. Right now we have different configs: the nevermined instance config which is also related to the network, and the nevermined config contract. It's important that the names don't raise confusion

File: src/commands/nfts/burnNft.ts [hunk 13]
```diff
@@ -50,13 +50,14 @@ export const burnNft = async (
       'nft721-sales'
     )
 
-    const nft = loadNftContract(config, nftAddress)
+    const nft = loadNftContract(config, nftAddress).connect(config.signer)
```
Comment (r-marques) [inline src/commands/nfts/burnNft.ts]: there is no need for a custom function for this. you can use `nvm.contracts.loadNft721(address)`. The sdk also has the erc-721 abi so there is no need to have it on the cli

File: src/commands/nfts/mintNft.ts [hunk 14]
```diff
@@ -55,13 +55,14 @@ export const mintNft = async (
       'nft721-sales'
     )
 
-    const nft = loadNftContract(config, nftAddress)
+    const nft = loadNftContract(config, nftAddress).connect(config.signer)
```
Comment (r-marques) [inline src/commands/nfts/mintNft.ts]: `nvm.contracs.loadNft721`

File: src/utils/CustomToken.ts [hunk 15]
```diff
@@ -1,28 +1,33 @@
...
     }
 
+    token.contract = loadERC20Contract(address, signer)
+    // token.contract = loadContract(config.config!, JSON.stringify(ERC20.abi), address)
+    // new ethers.Contract(address, abi)
     // @ts-ignore
-    token.contract = new token.web3.eth.Contract(abi, address)
-
+    // token.contract = new token.web3.eth.Contract(abi, address)
+    // console.log(token.contract.)
```
Comment (r-marques) [inline src/utils/CustomToken.ts]: cleanup

File: src/utils/CustomToken.ts [hunk 16]
```diff
@@ -1,28 +1,33 @@
...
     logger.info(`CustomToken with address ${address}`)
 
-    const code = await token.web3.eth.getCode(address)
+    const code = await token.web3.getCode(address)
     if (code === '0x0') {
       // no code in the blockchain dude
       throw new Error(`No code deployed at address ${address}, sorry.`)
     }
 
+    token.contract = loadERC20Contract(address, signer)
```
Comment (r-marques) [inline src/utils/CustomToken.ts]: here we can also use `nvm.contracts.loadErc20(address)`

File: src/utils/utils.ts [hunk 17]
```diff
@@ -34,7 +35,7 @@ export const loadNevermined = async (
     ...config.nvm,
     verbose: verbose ? verbose : config.nvm.verbose
   })
-
+  // console.log(`CONFIG NVM: ${JSON.stringify(config.nvm.web3Provider.)}`)
```
Comment (r-marques) [inline src/utils/utils.ts]: cleanup

File: src/utils/utils.ts [hunk 18]
```diff
@@ -73,6 +71,26 @@ export const loadNftContract = (
...
+    config.signer
+  )
+}
+
+export const loadERC20Contract = (
+  address: string,
+  signer: ethers.Signer
+): Contract => {
+  // const abi = `${ARTIFACTS_PATH}/NeverminedConfig.${config.networkName?.toLowerCase()}.json`
+  // const nvmConfigAbi = JSON.parse(fs.readFileSync(abiNvmConfig).toString())
```
Comment (r-marques) [inline src/utils/utils.ts]: cleanup

File: src/utils/utils.ts [hunk 19]
```diff
@@ -250,11 +309,16 @@ export const printNativeTokenBanner = async () => {
...
-    token.decimals(),
-    token.totalSupply()
+    token.contract.name(),
+    token.contract.symbol(),
+    token.contract.decimals(),
+    token.contract.totalSupply()
+    // token.name(),
+    // token.symbol(),
+    // token.decimals(),
+    // token.totalSupply()
```
Comment (r-marques) [inline src/utils/utils.ts]: cleanup

**Review by r-marques — APPROVED** [review]
lgtm

**Pull request merged**